### ISSUE TITLE
priority increase

### ISSUE THREAD

**Issue by sihyunjeon** [cms-PdmV/GridpackMachine#48]
Hello Geovanny,

The jobs I submitted two days ago are staying IDLE. If I look at McM dashboard, there are way too many PAG/POG 1/4/8 core validation jobs running and I think they are eating up higher priority (is PRI 2 given higher priority than 0 jobs?). These jobs I submitted are only allocated with 4 CPUs so in terms of "condor looking for job", they should be equivalent to validation jobs that uses 4 CPUs (maybe some differences in memory/storage requirements though).

Could we by default set the PRI = 3 in our machinery so that at least we get equal priority to other validation jobs? Current way, if we have 1000s of prepids being validated, our jobs won't run forever if we constantly have incoming prepids to validate.

Regards,
Sihyun.

==== copy pasted the email.

I think PRI=0 is the lowest priority and this will make our jobs stay IDLE forever if we have forever incoming prepids.
If PRI=0 is the lowest, can we elevate to 3 so that we don't lose priority to 8CPU prepid validation job?

**Comment by ggonzr** [comment]
Indeed, a higher value gives a higher [priority](https://htcondor.readthedocs.io/en/latest/classad-attributes/job-classad-attributes.html#JobPrio). Priority for McM validation jobs is given as shown [here](https://github.com/cms-PdmV/cmsPdmV/blob/9d10daa2dc3d2801d7a6041147e24bc3b3fbc104/mcm/automatic_scripts/validation/validation_control.py#L976-L997). 

My main concern would be the potential delay for McM validation jobs if someone submits a Gridpack job that requests 64 cores for instance. I propose the following to assign the priority for Gridpack jobs based on the requested number of cores:

```
cores <= 8 -> 3
8 < cores <= 16 -> 1
cores > 16 -> 0
```

Please let me know your thoughts about this proposal and if you agree with it

Best,
Geovanny

**Comment by sihyunjeon** [comment]
I would just ask for uniform priority

My understanding of scheduler was
1. there is a machine with 64 cores and it is running 20jobs, 44 cores are free
2. 64 core job wont run since the cores are not enough
3. Meanwhile smaller core jobs get submitted with lower priority
4. Smaller core jobs with lower priority sees the 44 cores machine and starts running

If this is true i dont think it matters

**Comment by sihyunjeon** [comment]
<img width="900" alt="image" src="https://github.com/cms-PdmV/GridpackMachine/assets/17442656/874382ee-a8b5-44cb-b4d5-f7f5cae10e6a">

**Comment by sihyunjeon** [comment]
if you look at this for example, 5664470.0 is running with 1 core before 5664468.0 with 4 cores, so it means the scheduler considers priority but also doesn't wait for a machine to be free when there is a higher priority job waiting

**Comment by sihyunjeon** [comment]
I guess your concern was the case like

1. a machine has 64 cores with some small jobs running
2. a job with PRI=3 with 64 core requirement gets submitted, stay in queue
3. 200 jobs with PRI=0 with 1 core requirement gets submitted, stay in queue
4. because of the PRI=3 job, machine does not hold 200 small jobs
5. when machine gets free and PRI=3 job first gets submitted
6. PRI=0 jobs wait for PRI=3 job to finish and starts running --> this will delay the whole process for validations

but based on observation how the jobs are being handled i think the machine does not handle jobs this way
instead,
5. PRI=0 jobs get submitted first, starts running jobs, 200 in total
6. after this is done, when the machine is free with 64 cores, the PRI=3 job starts running
